components/director/internal/open_resource_discovery/validation_client.go

```diff
@@ -0,0 +1,114 @@
+package ord
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	rulesetQueryParam = "ruleset"
+	validateEndpoint  = "/api/v1/document/validate"
+
+	contentTypeHeaderKey       = "Content-Type"
+	contentTypeApplicationJSON = "application/json;charset=UTF-8"
+)
+
+// ValidatorClient validates list of ORD documents with API Metadata Validator
+//
+//go:generate mockery --name=ValidatorClient --output=automock --outpkg=automock --case=underscore --disable-version-string
+type ValidatorClient interface {
+	Validate(ctx context.Context, ruleset, requestBody string) ([]ValidationResult, error)
+}
+
+// ValidationResult represents the structure of the response from the successful requests to API Metadata Validator
+type ValidationResult struct {
+	Code             string   `json:"code"`
+	Path             []string `json:"path"`
+	Message          string   `json:"message"`
+	Severity         string   `json:"severity"`
+	ProductStandards []string `json:"productStandards"`
+}
+
+// ValidationClient represents the client for the API Metadata Validator
+type ValidationClient struct {
+	url    string
+	client *http.Client
+}
+
+// NewValidationClient returns new validation client
+func NewValidationClient(url string, client *http.Client) *ValidationClient {
+	return &ValidationClient{
+		url:    url,
+		client: client,
+	}
+}
+
+// Validate sends request to API Metadata Validator to validate one ORD document
+func (vc *ValidationClient) Validate(ctx context.Context, ruleset string, requestBody string) ([]ValidationResult, error) {
+	if vc.url == "" {
```

Comment:
in my opinion it will be better to use the "enabled" property instead of relying on empty url string. 
If the metadata validator is disabled -> log and return empty validation errors
If the metadata validator is enabled -> use the provided host and port to build url. If this url is not correct, then fail.